client/src/backoffice/InboxListPage.tsx

```diff
@@ -0,0 +1,262 @@
+import React, { FC, useState, useEffect, ChangeEvent, useRef } from 'react';
+
+/* Libraries */
+import ReactModal from 'react-modal';
+import ReactPaginate from 'react-paginate';
+
+/* Helpers */
+import { Notification, getNotifications, getEvents, PoapEvent } from '../api';
+
+/* Components */
+import { Loading } from '../components/Loading';
+
+/* Assets */
+import plus from '../images/plus.svg';
+
+const PAGE_SIZE = 10;
+
+type PaginateAction = {
+  selected: number;
+};
+
+const InboxListPage: FC = () => {
+  const [page, setPage] = useState<number>(0);
+  const [total, setTotal] = useState<number>(0);
+  const [shouldResetPage, setShouldResetPage] = useState<boolean>(false);
+  const [notificationType, setNotificationType] = useState<string>('inbox');
+  const [recipientFilter, setRecipientFilter] = useState<string>('everyone');
+  const [modalOpen, setModalOpen] = useState<boolean>(false);
+  const [selectedEvent, setSelectedEvent] = useState<number | undefined>(undefined);
+  const [modalText, setModalText] = useState<string>('');
+  const [modalTitle, setModalTitle] = useState<string>('');
+  const [isFetchingNotifications, setIsFetchingNotifications] = useState<null | boolean>(null);
+  const [notifications, setNotifications] = useState<null | Notification[]>(null);
+  const [events, setEvents] = useState<PoapEvent[]>([]);
+
+  useEffect(() => {
+    fetchEvents();
+  }, []);
+
+  useEffect(() => {
+    page !== 0 ? setPage(0) : fetchNotifications();
+  }, [notificationType]);
+
+  useEffect(() => {
+    setShouldResetPage(true);
+  }, [recipientFilter, selectedEvent]);
+
+  useEffect(() => {
+    if (shouldResetPage) {
+      setShouldResetPage(false);
+
+      if (recipientFilter === 'event') {
+        if (selectedEvent === undefined) return;
+      } else {
+        setSelectedEvent(undefined);
+      }
+
+      page !== 0 ? setPage(0) : fetchNotifications();
+    }
+  }, [shouldResetPage]);
```

Comment:
My best guess to improve this code a little bit is to join some conditionals using AND operator. Also doing some early returns to avoid nested conditionals.

Code example:
```javascript
useEffect(() => {
  if (shouldResetPage === false) return;

  setShouldResetPage(false);

  if (recipientFilter === 'event' && selectedEvent === undefined) return;
  setSelectedEvent(undefined);

  page !== 0 ? setPage(0) : fetchNotifications();
}, [shouldResetPage]);
```